src/js/components/FormGroupComponent.jsx

```diff
@@ -2,17 +2,31 @@ var React = require("react/addons");
 
 var FormGroupComponent = React.createClass({
   displayName: "FormGroupComponent",
+
   propTypes: {
     attribute: React.PropTypes.string,
     children: React.PropTypes.object.isRequired,
     errors: React.PropTypes.array,
     help: React.PropTypes.string,
     label: React.PropTypes.string,
-    model: React.PropTypes.object.isRequired
+    model: React.PropTypes.object.isRequired,
+    validator: React.PropTypes.object.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      validationError: null
+    };
   },
 
   onInputChange: function (event) {
-    this.props.model.set(event.target.name, event.target.value);
+    var props = this.props;
+
+    props.model[event.target.name] = event.target.value;
```

Comment:
I am afraid we are doing it wrong here. `this.props.model` is passed down from `NewAppModalComponent`'s state, and I believe that in general props should be considered "immutable" in nature. Of course React allows us to do this, but I think we should use an internal input state:
```
getInitialState: function() {
  return {
    model: this.props.model
 ...
```